core/src/main/java/com/microsoft/applicationinsights/internal/shutdown/SDKShutdownActivity.java

```diff
@@ -51,10 +53,16 @@
     private static class SDKShutdownAction implements Runnable {
         private boolean stopped = false;
 
+        private final Map<TelemetryChannel, Boolean> channels = new WeakHashMap<>();
```

Comment:
I like this, but doesn't this mean when a `TelemetryClient` instance is GC'ed and its `TelemetryChannel` key is dropped that the channel's threads will still exist? If the `TelemetryClient` instance is being disposed, its `SDKShutdownAction` should be run.

Reply:
👍 Updated to non-weak